File: app/src/main/java/com/felipecsl/elifut/MatchStatistics.java
package com.felipecsl.elifut;

import android.support.annotation.Nullable;

import com.felipecsl.elifut.models.Club;
import com.felipecsl.elifut.models.Goal;

import org.apache.commons.math3.distribution.NormalDistribution;

import java.util.List;

public interface MatchStatistics {
  public static final float HOME_WIN_PROBABILITY = .465f;
  public static final float DRAW_PROBABILITY = HOME_WIN_PROBABILITY + .174f;
  public static final NormalDistribution GOALS_DISTRIBUTION = new NormalDistribution(2.6, 1.7);

  Club home();
  Club away();
  @Nullable Club winner();
  @Nullable Club loser();
  boolean isDraw();
  boolean isHomeWin();
  boolean isAwayWin();
  List<Goal> homeGoals();
  List<Goal> awayGoals();
  String finalScore();
  String refereeName();
  ClubStatistics homeStatistics();
  ClubStatistics awayStatistics();
}

File: app/src/main/java/com/felipecsl/elifut/MatchStatistics.kt
package com.felipecsl.elifut

import com.felipecsl.elifut.models.Club

import org.apache.commons.math3.distribution.NormalDistribution

interface MatchStatistics {
  fun home(): Club
  fun away(): Club
  fun winner(): Club?
  fun loser(): Club?
  fun isDraw(): Boolean
  fun isHomeWin(): Boolean
  fun isAwayWin(): Boolean
  fun homeGoals(): Int
  fun awayGoals(): Int
  fun finalScore(): String
  fun refereeName(): String
  fun homeStatistics(): ClubStatistics
  fun awayStatistics(): ClubStatistics

  data class ClubStatistics(val goals: Int, val shots: Int, val shotsOnTarget: Int, val corners: Int,
      val offsides: Int, val fouls: Int, val yellowCards: Int, val redCards: Int)

  companion object {
    val HOME_WIN_PROBABILITY = .465f
    val DRAW_PROBABILITY = HOME_WIN_PROBABILITY + .174f
    val GOALS_DISTRIBUTION = NormalDistribution(2.6, 1.7)
  }
}
